### PR TITLE
fix: Move all configuration to runtime from releases

### PR DESCRIPTION
**Pull request by arkadyan**
No ticket, this fixes issues I was encountering deploying to Dev and retrieving the API token secret.

Define SECRET_KEY_BASE for GitHub action.

Get env var on demand instead of on init to give the system time to
load it.

Also fixed another issue we noticed:
fix: Remove host setting from prod config
This is configured in runtime.

I was able to successfully [build](https://github.com/mbta/document_viewer/actions/runs/1262380281) and [deploy](https://github.com/mbta/document_viewer/actions/runs/1262395404) this branch to Dev.

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -1,5 +1,22 @@
 import Config
 
+if host = System.get_env("HOST") do
+  config :document_viewer, DocumentViewerWeb.Endpoint,
+    url: [host: host, port: 80],
+    secret_key_base: System.get_env("SECRET_KEY_BASE")
```
Comment (paulswartz) [inline config/runtime.exs:6]: question: does this part need to be part of the `HOST` if? or could it be separate?

Comment (arkadyan) [reply config/runtime.exs]: Nope, not only could it be separate, it should really have its own `if` statement to be safe. Fixed.

Comment (paulswartz) [reply config/runtime.exs]: I don't think you want it to have an `if` statement: you want the startup to crash if `SECRET_KEY_BASE` isn't set.

File: config/runtime.exs [hunk 2]
```diff
@@ -1,5 +1,22 @@
 import Config
 
+if host = System.get_env("HOST") do
+  config :document_viewer, DocumentViewerWeb.Endpoint, url: [host: host, port: 80]
+end
+
+config :document_viewer, DocumentViewerWeb.Endpoint, secret_key_base: secret_key_base
```
Comment (paulswartz) [inline config/runtime.exs:7]:
```suggestion
config :document_viewer, DocumentViewerWeb.Endpoint, secret_key_base: System.get_env("SECRET_KEY_BASE")
```

**Review by paulswartz — APPROVED** [review]
Looks good, pending the one suggestion. Give it another deploy to dev, and if that succeeds, 🍰

File: config/runtime.exs (hunk 2 above)
Comment (arkadyan) [reply config/runtime.exs]: Oops, moving too quickly. Thank you.

**Pull request merged**